src/daft-connect/src/spark_analyzer.rs

```diff
@@ -672,149 +689,213 @@ impl SparkAnalyzer<'_> {
-                let spark_expr::UnresolvedAttribute {
-                    unparsed_identifier,
-                    plan_id,
-                    is_metadata_column,
-                } = attr;
-
-                if let Some(plan_id) = plan_id {
-                    debug!(
-                        "Ignoring plan_id {plan_id} for attribute expressions; not yet implemented"
-                    );
-                }
-
-                if let Some(is_metadata_column) = is_metadata_column {
-                    debug!("Ignoring is_metadata_column {is_metadata_column} for attribute expressions; not yet implemented");
-                }
-
-                Ok(daft_dsl::unresolved_col(unparsed_identifier.as_str()))
-            }
-            spark_expr::ExprType::UnresolvedFunction(f) => self.process_function(f),
-            spark_expr::ExprType::ExpressionString(_) => {
-                not_yet_implemented!("Expression string not yet supported")
-            }
-            spark_expr::ExprType::UnresolvedStar(_) => {
-                not_yet_implemented!("Unresolved star expressions not yet supported")
-            }
-            spark_expr::ExprType::Alias(alias) => {
-                let spark_expr::Alias {
-                    expr,
-                    name,
-                    metadata,
-                } = &**alias;
-                let Some(expr) = expr else {
-                    invalid_argument_err!("Alias expression is required");
-                };
-
-                let [name] = name.as_slice() else {
-                    invalid_argument_err!("Alias name is required and currently only works with a single string; got {name:?}");
-                };
-
-                if let Some(metadata) = metadata {
-                    not_yet_implemented!("Alias metadata: {metadata:?}");
-                }
-
-                let child = self.to_daft_expr(expr)?;
-
-                let name = Arc::from(name.as_str());
-
-                Ok(child.alias(name))
-            }
-            spark_expr::ExprType::Cast(c) => {
-                let spark_expr::Cast {
-                    expr,
-                    eval_mode,
-                    cast_to_type,
-                } = &**c;
-
-                let Some(expr) = expr else {
-                    invalid_argument_err!("Cast expression is required");
-                };
-
-                let expr = self.to_daft_expr(expr)?;
-
-                let Some(cast_to_type) = cast_to_type else {
-                    invalid_argument_err!("Cast to type is required");
-                };
-
-                let data_type = match &cast_to_type {
-                    CastToType::Type(kind) => to_daft_datatype(kind)?,
-                    CastToType::TypeStr(s) => {
-                        not_yet_implemented!(
-                            "Cast to type string not yet supported; tried to cast to {s}"
-                        );
+        fn process_join_on(
```

Comment:
The implementation of this feels a little hacky but we will not need it after we implement non-equi joins anyway so I'm fine with keeping it as is.